src/main/scala/com/sksamuel/scapegoat/inspections/unneccesary/StoreBeforeReturn.scala

```diff
@@ -0,0 +1,41 @@
+package com.sksamuel.scapegoat.inspections.unneccesary
+
+import com.sksamuel.scapegoat._
+
+class StoreBeforeReturn
+    extends Inspection(
+      text = "Unnecessary store before return.",
+      defaultLevel = Levels.Info,
+      description = "Checks for storing a value in a block, and immediately returning the value.",
+      explanation =
+        "Storing a value and then immediately returning it is equivalent to returning the raw value itself."
+    ) {
+
+  override def inspector(context: InspectionContext): Inspector =
+    new Inspector(context) {
+      override def postTyperTraverser: context.Traverser =
+        new context.Traverser {
+          import context.global._
+
+          private def lastExprName(expr: Tree): Option[String] =
+            expr match {
+              case Return(Ident(name)) => Some(name.toString())
+              case Ident(name)         => Some(name.toString())
+              case _                   => None
+            }
+
+          override def inspect(tree: context.global.Tree): Unit =
+            tree match {
+              case DefDef(_, _, _, _, _, Block(stmts, lastExprInBody)) =>
+                val maybeLastExprName = lastExprName(lastExprInBody)
+                stmts.lastOption.foreach {
+                  case defn @ ValDef(_, assignmentName, _, _)
+                      if maybeLastExprName.contains(assignmentName.toString()) =>
+                    context.warn(defn.pos, self)
+                  case _ => stmts.foreach(inspect)
```

Comment:
I have a suspicion that this will blow up for mutually-recursive functions, but maybe the traversal algorithm already handles this? If not, any suggestions on how to approach this? I guess I could use an accumulator to keep track of visited locations, but that seems naive.

Reply:
> mutually-recursive functions,

Do you mean:
```

  def a(): Int = {
    val x = b()
    x
  }
  def b(): Int = {
    val x = a()
    x
  }
```
or something else?

> I have a suspicion [...] how to approach this?

Can you write down a UT for this case?
I am still not sure what's the suspected code.

Reply:
Added a unit test, looks like it isn't a problem (which makes sense, since this is just a syntactic check and the linter doesn't actually execute the code).

Reply:
Yes. We don't follow recursion (unless your inspection code chooses to, but even if it did it could only did for a single class / compilation unit)